### ISSUE TITLE
`no-conflicting-classes` reports incorrect error with `inset-ring` and `shadow-md`

### ISSUE THREAD

**Issue by tleunen** [sergioazoc/oxlint-tailwindcss#3]
Currently having this error when I apply `inset-ring-1` as well as `shadow-md`.

```
tailwindcss(no-conflicting-classes): "inset-ring-1" and "shadow-md" affect "box-shadow". "shadow-md" takes precedence (appears later).
```

`inset-ring-1` applies the following
```
.inset-ring-1 {
  --tw-inset-ring-shadow: inset 0 0 0 1px var(--tw-inset-ring-color, currentcolor);
  box-shadow: var(--tw-inset-shadow), var(--tw-inset-ring-shadow), var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow);
}
```
And `shadow-md` is doing this:
```
.shadow-md {
  --tw-shadow: 0 4px 6px -1px var(--tw-shadow-color, rgb(0 0 0 / 0.1)), 0 2px 4px -2px var(--tw-shadow-color, rgb(0 0 0 / 0.1));
  box-shadow: var(--tw-inset-shadow), var(--tw-inset-ring-shadow), var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow);
}
```

So we can see that both set `box-shadow` as the rule is saying, so that part is valid I guess. The problem though is that I cannot remove `inset-ring-1` because it also applies a shadow (`--tw-inset-ring-shadow`)that is then used by the `box-shadow` property.

So in this case, we must keep both of them, I don't think we can mix ring with shadow, as well as inset vs not.

**Comment by sergioazoc** [comment]
Fixed in v0.2.0.

The root cause was that `inset-ring-*` and `inset-shadow-*` classes weren't recognized as composing with `shadow-*` and `ring-*`. They all set `box-shadow` but via different CSS custom properties (`--tw-inset-ring-shadow`, `--tw-shadow`, etc.), so they're complementary, not conflicting.

The fix introduces a CSS custom property heuristic: if two classes both use `--tw-*` variables but none overlap, they are composing into a shared shorthand — not conflicting. This also covers other composition patterns that hadn't been reported yet: filter (`blur` + `brightness`), `backdrop-filter`, `contain`, `font-variant-numeric`, `touch-action`, `border-spacing`, and `mask`.

Same-utility conflicts (e.g. `shadow-sm shadow-lg`) are still correctly detected because they share the same custom property (`--tw-shadow`).
